// Copyright 2021 The Cross-Media Measurement Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package wfa_virtual_people;

import "src/main/proto/wfa/virtual_people/common/event.proto";
import "src/main/proto/wfa/virtual_people/common/label.proto";
import "src/main/proto/wfa/virtual_people/common/model.proto";

message LabelerInputList {
  repeated LabelerInput inputs = 1;
}

message LabelerOutputList {
  repeated LabelerOutput outputs = 1;
}

message CompiledNodeList {
  repeated CompiledNode nodes = 1;
}

message AggregatedReport {
  message Row {
    optional PersonLabelAttributes attrs = 1;
    optional int64 impressions = 2;
    optional int64 reach = 3;
  }

  repeated Row rows = 1;
}
